src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs

```diff
@@ -549,26 +550,20 @@ internal static void RemoveBreakHandler()
 
         private static readonly Lazy<ConsoleHandle> _keyboardInputHandle = new Lazy<SafeFileHandle>(() =>
             {
-                var handle = NativeMethods.CreateFile(
-                    "CONIN$",
-                    (UInt32)(NativeMethods.AccessQualifiers.GenericRead | NativeMethods.AccessQualifiers.GenericWrite),
-                    (UInt32)NativeMethods.ShareModes.ShareRead,
-                    (IntPtr)0,
-                    (UInt32)NativeMethods.CreationDisposition.OpenExisting,
-                    0,
-                    (IntPtr)0);
-
-                if (handle == NativeMethods.INVALID_HANDLE_VALUE)
+                ConsoleHandle handle = null;
+                try
                 {
-                    int err = Marshal.GetLastWin32Error();
-
-                    HostException e = CreateHostException(err, "RetreiveInputConsoleHandle",
-                                                            ErrorCategory.ResourceUnavailable,
-                                                            ConsoleControlStrings.GetInputModeExceptionTemplate);
+                    handle = File.OpenHandle("CONIN$", FileMode.Open, FileAccess.ReadWrite, FileShare.Read, FileOptions.None);
```

Comment:
Given the improvements in .NET's ReadKey() do we still need any of this anymore?  I realize this is a bigger change than what you intended here, but it would also remove much of this code.